src/LearnMe.Infrastructure/Repository/CalendarEventsRepository.cs

```diff
@@ -18,13 +18,23 @@ public CalendarEventsRepository(ApplicationDbContext context) : base(context)
             _context = context;
         }
 
+        public async Task<bool> DeleteByCalendarIdAsync(string calendarId)
+        {
+            var eventToBeDeleted = await GetByCalendarIdAsync(calendarId);
+
+            return await DeleteAsync(eventToBeDeleted.Id);
+        }
+
         public async Task<CalendarEvent> GetByCalendarIdAsync(string calendarId)
         {
             var result = await _context.CalendarEvents
                                                    .Where(o => o.CalendarId == calendarId)
                                                    .SingleOrDefaultAsync();
-            
-            _context.Entry(result).State = EntityState.Detached;
+
+            if (result != null)
+            {
+                _context.Entry(result).State = EntityState.Detached;

```

Comment:
zamiast czegoś takiego można napisać:

```csharp
var result = await _context.CalendarEvents 
    .Where(o => o.CalendarId == calendarId)
    .AsNoTracking() // as no tracking ma takie samo działanie jak detached                       
    .SingleOrDefaultAsync();               
```

Dzięki czemu cała funkcja może być krótsza:

```csharp
public async Task<CalendarEvent> GetByCalendarIdAsync(string calendarId)
{                                                                       
    return await _context.CalendarEvents                          
        .Where(o => o.CalendarId == calendarId)                         
        .AsNoTracking()                                                 
        .SingleOrDefaultAsync();                                                                                          
}                                                                       
````

Można to uprościć jeszcze bardziej. `Where` zwraca kolekcję. Dlaczego nie do razu znaleźć rządanego elementu? W metodzie oczekujesz jednego `CalendarEvent` czyli rozumiem, że calendarId może być tylko jedno w bazie. 

```csharp
public async Task<CalendarEvent> GetByCalendarIdAsync(string calendarId)
{                                                                       
    return await _context.CalendarEvents                                                 
        .AsNoTracking()                                                 
        .SingleOrDefaultAsync(o => o.CalendarId == calendarId); // raczej korzysta się z `x` zamiast `o` ale mniejsza                                                                                    
}                                                                       
````

Idąc dalej w optymalizacje można w ogole pominąć `await` bo twoja funkcja od razu zwraca result. Dlaczego to zadziała, długo by tłumaczyć. Możesz doczytać o multithreadingu w .necie

```csharp
public Task<CalendarEvent> GetByCalendarIdAsync(string calendarId)
{                                                                       
    return _context.CalendarEvents                                                 
        .AsNoTracking()                                                 
        .SingleOrDefaultAsync(o => o.CalendarId == calendarId); // raczej korzysta się z `x` zamiast `o` ale mniejsza                                                                                    
}                                                                       
````

Reply:
Oki super - doczytam. Ale zostawię sobie ten `await` dla pewności - przynajmniej na razie